### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by katareena**
Install and configure eslint-plugin-react-hooks;
Implement a possibility to view movie details;
use useState, useEffect, useContext

### CODE REVIEW

File: src/components/card/card.tsx [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+
+const Card: FunctionComponent<CardProps> = ({ movie }): JSX.Element => {
+  const { name, rating, runTime, genre, released, posterImage } = movie;
+
   return (
-    <section className='card visually-hidden'>
+    <section className='card'>
       <div className='card__inner'>
-        <div className='card__poster'></div>
+        <img className='card__poster' src={posterImage} alt='poster of movie' />
```
Comment (MarharytaKrasikava) [inline src/components/card/card.tsx:18]: I've found out that your page scrolls to the top all the time. The problem is in the footer component - `useEffect` gets triggered there on every component update. I didn't notice it in the previous review, sorry.

File: src/components/catalog-item/catalog-item.tsx [hunk 2]
```diff
@@ -16,16 +16,25 @@ const CatalogItem: FunctionComponent<CatalogItemProps> = ({
...
+    const movie: undefined | MovieType = movies.filter(
+      (movie) => movie.id === activeId
+    )[0];
+    setActiveMovie(movie);
+    setIsCardOpen(true);
+  }
 
   return (
-    <article className='catalog__card' onMouseOver={() => setActiveMovie(id)}>
+    <article className='catalog__card' onMouseOver={() => setActiveId(id)}>
```
Comment (MarharytaKrasikava) [inline src/components/catalog-item/catalog-item.tsx:32]: Why is it needed to manage this hovered item id thought the Context? `onMouseOver` you can just set the Edit menu visible with `setState()`.

**Review by MarharytaKrasikava — CHANGES_REQUESTED** [review]

File: src/hooks/context.tsx [hunk 3]
```diff
@@ -15,16 +23,54 @@ type ContextProps = {
...
+    released: '',
+  });
+
+  const fetchData = useCallback(() => {
+    setIsLoading(true);
+
+    setTimeout(() => {
+      try {
+        setMovies(mocksMovie);
+        setIsLoading(false);
```
[the remark this answers is not in the document]
Comment (katareena) [reply src/hooks/context.tsx:63]: done

File: src/components/header/header.tsx [hunk 4]
```diff
@@ -1,28 +1,34 @@
...
-            to={AppRoute.Signin}
-            style={{ marginLeft: 15, display: 'flex', alignItems: 'center' }}
-          >
-            LOGIN
-          </Link>
-        </div>
 
-        <FindButton />
+        {!isCardOpen ? (
+          <div style={{ display: 'flex' }}>
```
[the remark this answers is not in the document]
Comment (katareena) [reply src/components/header/header.tsx:20]: done

File: src/components/catalog/catalog.tsx [hunk 5]
```diff
@@ -37,19 +31,25 @@ const Catalog: FunctionComponent<CatalogProps> = ({ movies }): JSX.Element => {
...
-        {actualMovies.map((movie) => (
-          <CatalogItem movie={movie} key={movie.id} />
-        ))}
-      </div>
+      {isLoading ? (
+        <p className='page__loading'>Loading...</p>
+      ) : (
+        <>
+          <p className='catalog__result-text'>
+            <span>
```
[the remark this answers is not in the document]
Comment (katareena) [reply src/components/catalog/catalog.tsx:39]: done

File: src/components/catalog-item/catalog-item.tsx [hunk 6]
```diff
@@ -16,16 +16,25 @@ const CatalogItem: FunctionComponent<CatalogItemProps> = ({
...
   const { id, name, posterImage, genre, released } = movie;
-  const { activeMovie, setActiveMovie } = useGlobalContext();
-
-  console.log(activeMovie);
+  const { activeId, setActiveId, setIsCardOpen, movies, setActiveMovie } =
+    useGlobalContext();
+
+  function handlerClick(evt: React.MouseEvent<HTMLElement>) {
+    evt.preventDefault();
+    const movie: undefined | MovieType = movies.filter(
```
[the remark this answers is not in the document]
Comment (katareena) [reply src/components/catalog-item/catalog-item.tsx:24]: done

File: src/components/catalog-item/catalog-item.tsx [hunk 7]
```diff
@@ -16,16 +16,25 @@ const CatalogItem: FunctionComponent<CatalogItemProps> = ({
...
+      (movie) => movie.id === activeId
+    )[0];
+    setActiveMovie(movie);
+    setIsCardOpen(true);
+  }
 
   return (
-    <article className='catalog__card' onMouseOver={() => setActiveMovie(id)}>
+    <article className='catalog__card' onMouseOver={() => setActiveId(id)}>
       <Link
```
[the remark this answers is not in the document]
Comment (katareena) [reply src/components/catalog-item/catalog-item.tsx:33]: yes =) when switching to the card, we need routing /movies/${id},
in the following PR#6 this is  implemented

File: src/components/card/card.tsx [hunk 8]
```diff
@@ -1,24 +1,32 @@
...
+          <p className='card__genre'>{genre}</p>
 
           <p className='card__characteristic'>
-            <span>1994</span>
-            <span>2h 34min</span>
+            <span>{released}</span>
+            <span>{runTime}min</span>
           </p>
           <p className='card__description'>
             Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta)
```
[the remark this answers is not in the document]
Comment (katareena) [reply src/components/card/card.tsx:32]: done

File: src/components/card/card.tsx (hunk 1 above)
Comment (katareena) [reply src/components/card/card.tsx:18]: done

File: src/components/catalog-item/catalog-item.tsx (hunk 2 above)
Comment (katareena) [reply src/components/catalog-item/catalog-item.tsx:32]: done

**Pull request merged**